tests/unit/plugins/service/test_enricher.py

```diff
@@ -127,20 +162,52 @@ def gce_enricher(config, metrics, mock_http_client):
     (exceptions.GCPHTTPError('404 error'), 0, 1, 'GCPHTTPError: 404 error'),
     ([{}] * 5, 4, 5, 'KeyError: \'networkInterfaces\'')])
 @pytest.mark.asyncio
-async def test_handle_message_event_msg_failures(
-        mocker, gevent_msg, mock_async_sleep, gce_enricher, caplog, response,
-        sleep_calls, logs_logged, err_msg):
-    """Raise error while enriching event message."""
+async def test_handle_message_event_msg_additions_failures(
+        mocker, additions_gevent_msg, mock_async_sleep, gce_enricher, caplog,
+        response, sleep_calls, logs_logged, err_msg):
+    """Raise error while enriching event message with additions."""
     gce_enricher._http_client._get_json_mock.side_effect = response
-    gevent_msg.phase = 'enrich'
 
     with pytest.raises(exceptions.GCPGordonError) as e:
-        await gce_enricher.handle_message(gevent_msg)
+        await gce_enricher.handle_message(additions_gevent_msg)
 
-    assert 'enrich' == gevent_msg.phase
     assert sleep_calls == mock_async_sleep.call_count
     assert logs_logged == len(caplog.records)
     expected_msg = ('Could not get necessary information for '
                     'projects/123456789101/zones/us-central1-c/'
                     f'instances/an-instance-name-b34c: {err_msg}')
     assert e.match(expected_msg)
+
+
+@pytest.mark.asyncio
+async def test_handle_message_event_msg_deletions(
+        mocker, config, deletions_gevent_msg, gce_enricher, caplog,
+        records_data):
+    """Successfully enrich event message with records to delete."""
+    gce_enricher._http_client._get_all_mock.return_value = records_data
+
+    await gce_enricher.handle_message(deletions_gevent_msg)
+
+    expected_history_msg = 'Enriched msg with 2 resource record(s).'
+    assert expected_history_msg == \
+        deletions_gevent_msg.history_log[0]['message']
+    assert 'enrich' == deletions_gevent_msg.history_log[0]['plugin']
```

Comment:
not enricher?

Reply:
I think it should be `enricher`. We set it to `enrich` here: https://github.com/spotify/gordon-gcp/blob/07a39c411ba0fc79f7407a6372ed7f901f479fca/src/gordon_gcp/plugins/service/enricher.py#L128, but since it's `plugin`, `enricher` probably makes more sense. I think `enricher` as a phase is also acceptable nomenclature.

Side note that we're not appending this history bit to the gevent_msg at all in the `gdns_publisher`. Is that on purpose ?

Reply:
I don't know, but I think we should.

Reply:
Looks like the phases are verbs, not plugin names, after all.  So enrich is correct unless we want to change them everywhere.

Reply:
I'll create an issue for it for later. I don't think we should try to stuff it into this PR.